src/package1/CustomArrayList.java

```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
 
     @Override
     public Iterator<String> iterator() {
-        return null;
+        throw new UnsupportedOperationException();
     }
 
-    public static Object[] toArray(List<String> name) {
-
-        return name.toArray(new String[name.size()]);
+    public Object[] toArray() {
+        return Arrays.copyOf(elements, size);
     }
 
     @Override
     public <T> T[] toArray(T[] a) {
-        return null;
+        throw new UnsupportedOperationException();
     }
 
     @Override
-    public boolean add(List<String> name, String s) {
-        String [] a = new String[name.size() + 1];
+    public boolean add(String s) {
+
+        elements[size] = s;
+        size++;
 
-        for (int i = 0; i < name.size(); i++) {
-            a[i] = name.get(i);
+        if (size == elements.length) {
+            String[] a = new String[(int) (elements.length * 1.5)];
+            elements = a;
         }
-        a[name.size() + 1] = s;
-        Arrays.asList(a);
+
         return true;
     }
 
+//    Використовувати size замість length
+//    Елементи мають переміщуватись вліво в сторону видаленого елемента.
     @Override
-    public boolean remove(List<String> name, Object o) {
-        for (int i = 0; i < name.size(); i++) {
-            if (name.get(i).equals(o)) {
-                name.remove(name.get(i));
+    public boolean remove(Object o) {
```

Comment:
1. Use size, not length
2. Remove is just shift elements to left

Reply:
Fixed.